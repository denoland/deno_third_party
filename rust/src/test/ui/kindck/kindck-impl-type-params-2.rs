#![feature(box_syntax)]

trait Foo {
}

impl<T:Copy> Foo for T {
}

fn take_param<T:Foo>(foo: &T) { }

fn main() {
    let x: Box<_> = box 3;
    take_param(&x);
    //~^ ERROR `std::boxed::Box<{integer}>: std::marker::Copy` is not satisfied
}
