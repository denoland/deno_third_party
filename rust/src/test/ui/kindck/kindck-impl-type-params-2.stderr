error[E0277]: the trait bound `std::boxed::Box<{integer}>: std::marker::Copy` is not satisfied
  --> $DIR/kindck-impl-type-params-2.rs:13:5
   |
LL |     take_param(&x);
   |     ^^^^^^^^^^ the trait `std::marker::Copy` is not implemented for `std::boxed::Box<{integer}>`
   |
   = note: required because of the requirements on the impl of `Foo` for `std::boxed::Box<{integer}>`
note: required by `take_param`
  --> $DIR/kindck-impl-type-params-2.rs:9:1
   |
LL | fn take_param<T:Foo>(foo: &T) { }
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
